### PR TITLE
hal: Add HAL for the CCM peripheral

### PR DESCRIPTION
**Pull request by anangl**
This is a continuation of the work started by @aescolar-ot and replaces his PR #26.

Added functions for accessing all remaining registers in the CCM peripheral to provide a complete solution.

### CODE REVIEW

File: hal/nrf_ccm.h [hunk 1]
```diff
@@ -0,0 +1,495 @@
...
+ */
+__STATIC_INLINE void nrf_ccm_disable(NRF_CCM_Type * p_reg);
+
+/**
+ * @brief Function for setting the CCM peripheral configuration.
+ *
+ * @param[in] p_reg    Pointer to the peripheral registers structure.
+ * @param[in] p_config Pointer to the structure with configuration to be set.
+ */
+__STATIC_INLINE void nrf_ccm_configure(NRF_CCM_Type *          p_reg,
```
Comment (kl-cruz) [inline hal/nrf_ccm.h]: Apply one consistent style of indentation in file. There is no args indentation in previous lines but after this one - there is.

File: hal/nrf_ccm.h [hunk 2]
```diff
@@ -0,0 +1,495 @@
...
+
+#ifndef SUPPRESS_INLINE_IMPLEMENTATION
+
+__STATIC_INLINE void nrf_ccm_task_trigger(NRF_CCM_Type * p_reg, nrf_ccm_task_t task)
+{
+    *((volatile uint32_t *)((uint8_t *)p_reg + (uint32_t)task)) = 0x1UL;
+}
+
+__STATIC_INLINE uint32_t nrf_ccm_task_address_get(NRF_CCM_Type const * p_reg,
+                                                  nrf_ccm_task_t task)
```
Comment (kl-cruz) [inline hal/nrf_ccm.h]: Indentation. In line 342 code was aligned

File: hal/nrf_ccm.h [hunk 3]
```diff
@@ -0,0 +1,495 @@
...
+{
+    p_reg->ENABLE = (CCM_ENABLE_ENABLE_Enabled << CCM_ENABLE_ENABLE_Pos);
+}
+
+__STATIC_INLINE void nrf_ccm_disable(NRF_CCM_Type * p_reg)
+{
+    p_reg->ENABLE = (CCM_ENABLE_ENABLE_Disabled << CCM_ENABLE_ENABLE_Pos);
+}
+
+__STATIC_INLINE void nrf_ccm_configure(NRF_CCM_Type *          p_reg,
```
Comment (kl-cruz) [inline hal/nrf_ccm.h]: add one space and will be great!

File: hal/nrf_ccm.h [hunk 4]
```diff
@@ -0,0 +1,495 @@
...
+ * @brief   Hardware access layer for managing the AES CCM peripheral.
+ */
+
+/**
+ * @brief CCM tasks.
+ */
+typedef enum
+{
+    /*lint -save -e30*/
+    NRF_CCM_TASK_KSGEN        = offsetof(NRF_CCM_Type, TASKS_KSGEN),        ///< Start generation of key-stream. */
```
Comment (kl-cruz) [inline hal/nrf_ccm.h]: I think this kind of doc comment should not be finished with */

**Comment by anangl** [comment]
@kl-cruz I addressed your comments.

**Comment by anangl** [comment]
After an offline discussion with @kl-cruz I changed the title of the first commit to make it consistent with all the rest. @aescolar-ot I hope you don't mind.

**Comment by aescolar-ot** [comment]
@anangl of course it's ok :)

File: hal/nrf_ccm.h [hunk 5]
```diff
@@ -0,0 +1,497 @@
...
+
+__STATIC_INLINE void nrf_ccm_disable(NRF_CCM_Type * p_reg)
+{
+    p_reg->ENABLE = (CCM_ENABLE_ENABLE_Disabled << CCM_ENABLE_ENABLE_Pos);
+}
+
+__STATIC_INLINE void nrf_ccm_configure(NRF_CCM_Type *           p_reg,
+                                       nrf_ccm_config_t const * p_config)
+{
+    p_reg->MODE = ((p_config->mode     << CCM_MODE_MODE_Pos) |
```
Comment (kl-cruz) [inline hal/nrf_ccm.h]: Consider to cast those enums to uint32_t to make it safe when working with for example -fshort-enums in gcc

**Comment by anangl** [comment]
@kl-cruz Updated documentation of `nrf_ccm_scratchptr_set` and added documentation bindings. Could you take another look?

**Comment by kl-cruz** [comment]
Sure!

**Comment by kl-cruz** [comment]
Looks good now. Ready for merge

**Pull request merged**